openshift-java/Jenkinsfile

```diff
@@ -0,0 +1,58 @@
+#!/usr/bin/env groovy
+
+node('rhel7'){
+	stage('Checkout repo') {
+		deleteDir()
+		git url: 'https://github.com/redhat-developer/vscode-openshift-extension-pack/'
+        dir: 'openshift-java'
```

Comment:
I don't think git step can actually checkout a subdirectory (looking at https://jenkins.io/doc/pipeline/steps/git/). My guess is it's just going to checkout the whole repo and then fail on the commands that need package.json in their work directory.

Reply:
@jrichter1 Do we have any other way to checkout a specifc dir, as this is how we have the structure for specifc extension pack.

Reply:
If there is such functionality, I guess it would be in the 'checkout' command (https://jenkins.io/doc/pipeline/steps/workflow-scm-step/), but I can't find anything like that. But you could checkout the entire repo and manually change into the desired folder.
 - wrapping commands in 'dir': dir('myFolder') { code } where the code would have its pwd set to myFolder (relative to workspace root)
 - run a shell script to isolate the folder you want (i.e. move everything from the folder to workspace root and delete unnecessary folders, the possibilities are endless)